File: 0009.py
import math

def brute_force():
    a = 100
    b = 100
    while True:
        if b > a:
            b = 100
            a += 1
        c = math.sqrt((a ** 2) + (b ** 2))
        if a + b + c == 1000:
            print a, '*', b, '*', c
            return a * b * c
        b += 1


def fn0():
    '''

    a + b + c = 1000
    a ** 2 + b ** 2 = c ** 2


    '''

    return brute_force()


solutions = [brute_force]

File: 0009.c
#include <stdio.h>
#include <math.h>

void fn0() {
  int a = 100;
  int b = 100;
  while (1) {
    if (b > a) {
      b = 100;
      a += 1;
    }
    double c = sqrt(a * a + b * b);
    if (fmod(c, 1.0) == 0 && a + b + (int)c == 1000) {
      printf("%d\n",  a * b * (int)c);
      return;
    }
    b += 1;
  }
}

void (*solutions[])() = { fn0, NULL };
